src/main/src/main/java/org/geoserver/ows/StylePublisher.java

```diff
@@ -0,0 +1,123 @@
+/* (c) 2014 Open Source Geospatial Foundation - all rights reserved
+ * (c) 2001 - 2013 OpenPlans
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.ows;
+
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLDecoder;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.geoserver.catalog.Catalog;
+import org.geoserver.catalog.WorkspaceInfo;
+import org.geoserver.config.GeoServerDataDirectory;
+import org.geoserver.platform.GeoServerResourceLoader;
+import org.geoserver.platform.resource.Resource;
+import org.geoserver.platform.resource.Resource.Type;
+import org.geotools.data.DataUtilities;
+import org.springframework.web.context.support.ServletContextResourceLoader;
+
+/**
+ * Controller which publishes styles through a web interface.
+ * <p>
+ * To use this controller, it should be mapped to a particular url in the url mapping of the spring dispatcher servlet.
+ * 
+ * @author Alex Goudine, Boundless
+ */
+public class StylePublisher extends AbstractURLPublisher {
+    /**
+     * Resource loader
+     */
+    protected Catalog catalog;
+
+    /**
+     * Servlet resource loader
+     */
+    protected ServletContextResourceLoader scloader;
+
+    /**
+     * Creates the new file publisher.
+     * 
+     * @param loader The loader used to locate files.
+     */
+    public StylePublisher(Catalog catalog) {
+        this.catalog = catalog;
+    }
+
+    // @Override
+    // protected void initServletContext(ServletContext servletContext) {
+    // this.scloader = new ServletContextResourceLoader(servletContext);
+    // }
+
+    @Override
+    protected URL getUrl(HttpServletRequest request) throws IOException {
+        String ctxPath = request.getContextPath();
+        String reqPath = request.getRequestURI();
+        reqPath = URLDecoder.decode(reqPath, "UTF-8");
+        reqPath = reqPath.substring(ctxPath.length());
+
+        if ((reqPath.length() > 1) && reqPath.startsWith("/")) {
+            reqPath = reqPath.substring(1);
+        }
+
+        // sigh, in order to serve the file we have to open it 2 times
+        // 1) to determine its mime type
+        // 2) to determine its encoding and really serve it
+        // we can't coalish 1) because we don't have a way to give jmimemagic the bytes at the
+        // beginning of the file without disabling extension quick matching
```

Comment:
I do not see evidence of opening the file in the following code? Is this comment up to date?